integration/spark2/src/main/java/org/apache/carbondata/spark/vectorreader/VectorizedCarbonRecordReader.java

```diff
@@ -116,6 +117,14 @@ public VectorizedCarbonRecordReader(QueryModel queryModel, InputMetricsStats inp
       queryExecutor = QueryExecutorFactory.getQueryExecutor(queryModel);
       iterator = (AbstractDetailQueryResultIterator) queryExecutor.execute(queryModel);
     } catch (QueryExecutionException e) {
+      Throwable ext = e;
+      while (ext != null) {
+        if (ext instanceof FileNotFoundException) {
+          throw new InterruptedException(
```

Comment:
better pass the exception along with message to the constructor

Reply:
InterruptedException takes only one parameter that's only message